libsplinter/src/peer/token.rs

```diff
@@ -65,7 +67,9 @@ impl PeerAuthorizationToken {
     pub fn has_public_key(&self, public_keys: &[Vec<u8>]) -> bool {
         match self {
             PeerAuthorizationToken::Trust { .. } => false,
-            PeerAuthorizationToken::Challenge { public_key } => public_keys.contains(&public_key),
+            PeerAuthorizationToken::Challenge { public_key } => {
+                public_keys.contains(&public_key.as_slice().to_vec())
```

Comment:
Why does this need to be converted to a vec?  There might be a way to do this without an allocation.

Reply:
This method is actually not used anywhere so I added a commit to remove it